components/usage/pkg/stripe/stripe.go

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) 2022 Gitpod GmbH. All rights reserved.
+// Licensed under the GNU Affero General Public License (AGPL).
+// See License-AGPL.txt in the project root for license information.
+
+package stripe
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/gitpod-io/gitpod/common-go/log"
+	"github.com/stripe/stripe-go/v72"
+	"github.com/stripe/stripe-go/v72/customer"
+)
+
+type stripeKeys struct {
+	PublishableKey string `json:"publishableKey"`
+	SecretKey      string `json:"secretKey"`
+}
+
+func Authenticate(apiKeyFile string) error {
+	bytes, err := os.ReadFile(apiKeyFile)
+	if err != nil {
+		return err
+	}
+
+	var stripeKeys stripeKeys
+	err = json.Unmarshal(bytes, &stripeKeys)
+	if err != nil {
+		return err
+	}
+
+	stripe.Key = stripeKeys.SecretKey
+	return nil
+}
+
+// FindCustomersForTeamIds queries the stripe API to find all customers with a teamId in `teamIds`.
+func FindCustomersForTeamIds(teamIds []string) {
+	queries := queriesForCustomersWithTeamIds(teamIds)
+
+	for _, query := range queries {
+		log.Infof("about to make query %q", query)
+		params := &stripe.CustomerSearchParams{SearchParams: stripe.SearchParams{Query: query}}
+		iter := customer.Search(params)
+		for iter.Next() {
+			customer := iter.Customer()
+			log.Infof("found customer %q for teamId %q", customer.Name, customer.Metadata["teamId"])
+		}
+	}
+}
+
+// queriesForCustomersWithTeamIds constructs Stripe query strings to find the Stripe Customer for each teamId
+// It returns multiple queries, each being a big disjunction of subclauses so that we can process multiple teamIds in one query.
+// `clausesPerQuery` is a limit enforced by the Stripe API.
+func queriesForCustomersWithTeamIds(teamIds []string) []string {
+	const clausesPerQuery = 10
```

Comment:
Interesting. I guess you found this limit by testing. Is the value `10` documented somewhere? 👀

Reply:
Not documented anywhere I could find. It appeared in an error message when trying larger queries.

Reply:
Yes, I wondered about differences between live vs staging too.

We will likely need to rate limit our Stripe API calls in production in general. For this search endpoint the rate limit is:

> Rate limits
We apply a [rate limit](https://stripe.com/docs/rate-limits) of up to 20 read operations per second which applies for all search endpoints in both live mode and test mode. Live mode and test mode limits are separate.